cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetTest.java

```diff
@@ -143,6 +146,20 @@ public void testDecodeIncomplete() {
   }
 
   @Test
+  public void testMetadataForNonexistentPartition() throws Exception {
+    PartitionedFileSet pfs = dsFrameworkUtil.getInstance(pfsInstance);
+    PartitionKey key = generateUniqueKey();
+    try {
+      // didn't add any partitions to the dataset, so any partition key should throw a PartitionNotFoundException
+      pfs.addMetadata(key, "metaKey", "metaValue");
+      Assert.fail("Expected not to find key: " + key);
+    } catch (PartitionNotFoundException expected) {
+      Assert.assertEquals(pfsInstance.getId(), expected.getPartitionedFileSetName());
```

Comment:
expected should come first, ``Assert.assertEquals(expected.getPartitionedFileSetName(), pfsInstance.getId());`` same with the next assertion.